distributions/gcp/Kptfile

```diff
@@ -850,3 +850,27 @@ openAPI:
           values:
           - marker: ${mgmt-project}
             ref: '#/definitions/io.k8s.cli.setters.mgmt-project'
+    io.k8s.cli.setters.node-locations:
+      type: array
+      x-k8s-cli:
+        setter:
+          name: node-locations
+          listValues: ["ZONE"]
+    io.k8s.cli.setters.enable-private-endpoint:
+      type: string
+      x-k8s-cli:
+        setter:
+          name: enable-private-endpoint
+          value: "false"
+    io.k8s.cli.setters.master-ip-cidr-block:
+      type: string
+      x-k8s-cli:
+        setter:
+          name: master-ip-cidr-block
+          value: 172.16.0.32/28
+    io.k8s.cli.setters.master-global-access-enabled:
```

Comment:
Sorry I don't have enough experience with private clusters, are this config and enable private endpoint config reasonable for both enabled or disabled per use case?
Can you briefly explain when they should be set the non default values?

Thank you!

Reply:
enable-private-endpoint is false, by default, making the control plane operating from a public endpoint. One can set a masterAuthorizedNetworksConfig list of CIDRs to authorize on that public endpoint, or one can enable the private endpoint, and have accessibility from things in the same VPC. master-global-access-enabled is false, by default, and setting this to true along with enable-private-endpoint=true will make the control plane accessible outside of the VPC to ranges set in masterAuthorizedNetworksConfig. 

I'm not sure how to make masterAuthorizedNetworksConfig configurable with kustomize/kpt and would just add my own separate patch for this. Perhaps master-global-access should be more tied to that patch than this patch, because you'd likely only enable global-access to a specific network range you'd specify in masterAuthorizedNetworksConfig.